khook/engine.c

```diff
@@ -116,13 +120,16 @@ int khook_init(void)
 	// have eXecutable attributes. That's why we have to mark the
 	// region executable explicitly.
 	//
-
 	set_memory_x = (void *)khook_lookup_name("set_memory_x");
-	if (set_memory_x) {
-		int numpages = round_up(KHOOK_STUB_TBL_SIZE, PAGE_SIZE) / PAGE_SIZE;
+#ifdef  USE_CR0_FAM
+	if (set_memory_x)
 		set_memory_x((unsigned long)khook_stub_tbl, numpages);
```

Comment:
Why we can't leave `set_memory_x`  as is?

Reply:
we can. but, It can be catch via monitoring systems, for example.

Reply:
`set_memory_x` is based on PTE as well so I don't see a problem

Reply:
ok, but my implementation was more pretty